packages/react/widget/src/hooks/use-portal-widgets.ts

```diff
@@ -0,0 +1,59 @@
+import { useMemo, useRef, useState } from 'react';
+import { createElement } from '../util/element';
+import { widgetRender } from '../render/render';
+import { IWidgetModuleProvider, WidgetProps } from '@equinor/fusion-framework-module-widget';
+import { Fusion } from '@equinor/fusion-framework';
+
+export const usePortalWidgets = <TProps extends WidgetProps>(
+    provider: IWidgetModuleProvider,
+    cb: () => {
+        name: string;
+        props?: TProps;
+        fusion: Fusion;
+        widgetVersion?: {
+            type: 'version' | 'tag';
+            value: string;
+        };
+    },
+) => {
+    const [loading, setLoading] = useState(false);
+    const [error, setError] = useState<Error | undefined>();
+
+    const widgetRef = useRef<HTMLDivElement>(createElement());
+    const { name, props, widgetVersion, fusion } = cb();
```

Comment:
why is the input a cb, when it's resolved on every call?

Reply:
You are right, will fix ;) shall i call if config?